subsys/bluetooth/host/hci_core.c

```diff
@@ -670,6 +674,30 @@ static int hci_le_read_remote_features(struct bt_conn *conn)
 	return 0;
 }
 
+#if defined(CONFIG_BT_BLUENRG_ACI)
+static int hci_send_aci_config_data_controller_mode(void)
```

Comment:
Ideally this kind of stuff should be completely on the driver side. Does HCI_Reset undo the effects of the command you're sending? If not, you could do the command sending in the drivers's open() callback.

Reply:
The command MUST be sent just after HCI_reset to have an effect on bluetooth component internal stack and it's not possible change the role of the internal stack before or after (much later on time) HCI_reset

Reply:
Is HCI_Reset; Vendor_command; HCI_Reset; also something that will not work? Will the second HCI_Reset throw the controller's internal host stack back on? If that's also not an option, we could add a new optional callback to HCI drivers (called e.g. "setup") that's called after the host has sent the first HCI_Reset. That way the hardware-specific init would still stay inside the HCI driver.

Reply:
no, it's only HCI_Reset; Vendor_command.

Reply:
Ok, in that case I think the most reasonable thing to do is an optional driver->setup() callback after hci_core.c has sent HCI_Reset. Btw, where can we find reference documentation to this vendor command, its parameters, and how the command relates to HCI_Reset? Is that information available somewhere publicly?

We're planning to add a bt_disable() to accompany bt_enable() at some point in the future, and that will most likely involve calling HCI_Reset again (when bt_enable() is called a second time), so we need to make sure that no behavior is broken.

Reply:
Ok, the way you've cleaned this up with appropriate defines and a parameter struct looks good. However, it'd still be good to move all of it over to `drivers/bluetooth/hci/spi.c` with the help of a new optional HCI driver callback. I've got a patch ready for this, but am a little hesitant since I'm not sure how easily this will be reusable for other vendor extensions (e.g. the exact order of HCI_Reset vs vendor commands), and how we should then deal with other drivers that require a different order.

Was it so that HCI_Reset essentially resets all vendor settings as well? Will it be possible to do HCI_Reset; vendor_write_config; HCI_Reset; vendor_write_config? This is important information for introducing a bt_disable() API (something that may happen already for Zephyr 1.13)

Reply:
@jhedberg I have made the test for HCI_Reset; vendor_write_config; HCI_Reset; vendor_write_config  and it doesn't work, my internal blueNRG stack are not on a stable state after this list of command. With my test with ipsp sample, only HCI_Reset; vendor_write_config (make one time) work correctly to perform the ipsp use cases.